phoenix-core/src/it/java/org/apache/phoenix/end2end/AlterAddCascadeIndexIT.java

```diff
@@ -0,0 +1,272 @@
+                    "us_population_gv (avg_fam_size) INCLUDE (population)");
+            databaseObjectName = "us_population_gv";
+            indexesName = "us_population_gv_gi, us_population_gv_gi_2";
+            planOutput = isLocalIndex? "US_POPULATION" : "_IDX_US_POPULATION";
+
+        } else {
+            conn.createStatement().execute("CREATE "+ (isLocalIndex ? "LOCAL " : "") + "INDEX IF NOT EXISTS us_population_gi ON " +
+                    "us_population (population)");
+            conn.createStatement().execute("CREATE "+ (isLocalIndex ? "LOCAL " : "") + "INDEX IF NOT EXISTS us_population_gi_2 ON " +
+                    "us_population (state, population)");
+            databaseObjectName = "us_population";
+            indexesName = "us_population_gi, us_population_gi_2";
+            planOutput = isLocalIndex? "US_POPULATION" : "us_population_gi";
+        }
+    }
+
+    private void loadData() throws SQLException {
+        PreparedStatement ps = conn.prepareStatement("UPSERT INTO us_population VALUES('NY','New York',8143197)");
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('CA','Los Angeles',3844829)");
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('IL','Chicago',2842518)");
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('TX','Houston',2016582)");
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('PA','Philadelphia',1463281)");
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('AZ','Phoenix',1461575)");
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('TX','San Antonio',1256509)");
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('CA','San Diego',1255540)");
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('TX','Dallas',1213825)");
+        ps.executeUpdate();
+        ps = conn.prepareStatement("UPSERT INTO us_population VALUES('CA','San Jose',912332)");
+        ps.executeUpdate();
+    }
+
+    // Test with ALTER TABLE CASCADE INDEX ALL
+    @Test
+    public void testAlterDBOAddCascadeIndexAll() throws SQLException {
+        String query = "ALTER " +(isViewIndex? "VIEW " : "TABLE ") + databaseObjectName +" ADD new_column VARCHAR CASCADE INDEX ALL";
+        conn.createStatement().execute(query);
+        query = "EXPLAIN SELECT new_column FROM "+databaseObjectName;
+        ResultSet rs = conn.prepareStatement(query).executeQuery();
+        rs.next();
+        //confirm it is using global index to access the column
+        Assert.assertTrue(rs.getString(1).contains(planOutput.toUpperCase()));
```

Comment:
In fact, it would be easier if you have separate variables like globalIndexPlanOutput. It is easier to see what you are comparing against that way. If you add another test and just have planOutput comparison without a comment on top, it would be difficult to understand what you are asserting for.